### PR TITLE
`ClientOptionsChange` paper-only event

### PR DESCRIPTION
**Pull request by heypr**
- Requested by [Mergu](https://discord.com/channels/315163488085475337/1209708259859234868) on Discord.

### CODE REVIEW

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        return switch (name) {
+            case "server_listings_enabled" -> NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19) ? new ElementTag(event.allowsServerListings()) : null;
+            case "chat_visibility" -> new ElementTag(event.getChatVisibility());
+            case "locale" -> new ElementTag(event.getLocale());
+            case "main_hand" -> new ElementTag(event.getMainHand());
+            case "skin_parts" -> {
+                MapTag map = new MapTag();
+                map.putObject("cape", new ElementTag(event.getSkinParts().hasCapeEnabled()));
+                map.putObject("jacket", new ElementTag(event.getSkinParts().hasJacketEnabled()));
+                map.putObject("leftSleeve", new ElementTag(event.getSkinParts().hasLeftSleeveEnabled()));
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java]: `left_sleeve`

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    // @Player Always.
+    //
+    // @Triggers when a player changes their client options.
+    //
+    // @Context
+    // <context.server_listings_enabled> returns an ElementTag(Boolean) of whether the player has server listings enabled. Available only on MC 1.19+.
+    // <context.chat_visibility> returns the player's chat visibility.
+    // <context.locale> returns the player's locale.
+    // <context.main_hand> returns an ElementTag(Boolean) of whether the player's main hand has changed.
+    // <context.skin_parts> returns a MapTag of whether the player's skin parts are enabled or not.
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java]: Should have an example or key list

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    @Override
+    public ObjectTag getContext(String name) {
+        return switch (name) {
+            case "server_listings_enabled" -> NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19) ? new ElementTag(event.allowsServerListings()) : null;
+            case "chat_visibility" -> new ElementTag(event.getChatVisibility());
+            case "locale" -> new ElementTag(event.getLocale());
+            case "main_hand" -> new ElementTag(event.getMainHand());
+            case "skin_parts" -> {
+                MapTag map = new MapTag();
+                map.putObject("cape", new ElementTag(event.getSkinParts().hasCapeEnabled()));
```
Comment (mcmonkey4eva) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java]: grab the `getSkinParts()` return and put it in a variable, so you're not calling the same `getSkinParts` method 7 times in a row

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    //
+    // @Player Always.
+    //
+    // @Triggers when a player changes their client options.
+    //
+    // @Context
+    // <context.server_listings_enabled> returns an ElementTag(Boolean) of whether the player has server listings enabled. Available only on MC 1.19+.
+    // <context.chat_visibility> returns the player's chat visibility.
+    // <context.locale> returns the player's locale.
+    // <context.main_hand> returns an ElementTag(Boolean) of whether the player's main hand has changed.
```
Comment (tal5) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java]: Doesn't sound right? this should be a `LEFT`/`RIGHT` enum

File: paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    // <context.skin_parts> returns a MapTag of whether the player's skin parts are enabled or not. For example: [cape=true;jacket=false;left_sleeve=true;right_sleeve=false;left_pants=true;right_pants=false;hat=true]
+    // <context.view_distance> returns the player's view distance.
+    // <context.server_listings_changed> returns whether the player's server listings have changed. Available only on MC 1.19+.
+    // <context.chat_colors> returns whether the player has chat colors enabled.
+    // <context.chat_colors_changed> returns whether the player's chat colors have changed.
```
Comment (tal5) [inline paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java]: This (and a few other `_changed` meta) seem a bit unclear to me - it's not that the client has a chat color and that changes, it's whether the client allows chat colors that changes

**Review by mcmonkey4eva — APPROVED** [review]

**Review by tal5 — APPROVED** [review]

**Pull request merged**